lightning/src/routing/scoring.rs

```diff
@@ -2548,32 +2553,40 @@ mod tests {
 			inflight_htlc_msat: 0,
 			effective_capacity: EffectiveCapacity::Total { capacity_msat: 1_024_000, htlc_maximum_msat: 1_000 },
 		};
-		assert_eq!(scorer.channel_penalty_msat(42, &source, &target, usage, &params), 0);
+		let channel = network_graph.read_only().channel(42).unwrap().to_owned();
```

Comment:
In some places I was able to take out the `to_owned` but in others, I couldnt as I got an error saying the lock is not free. how is that possible when we are using `RwLock`, cant multiple functions read from the lock in the same time?

Reply:
You'll need something like this:

```rust
let network_graph = network_graph.read_only();
let channel = network_graph.channel(42).unwrap();
```

Hard to say why it would sometimes work without the extra variable without seeing an example.